tests/_test_base.bash

```diff
@@ -239,7 +256,13 @@ function set_state_initial {
 
 
 function set_state_git {
-  git init >> "$TEST_OUTPUT_FILE" 2>&1
+  local has_initial_branch_option
+  has_initial_branch_option=$(is_git_version_ge_2_28_0) # 0 for true
+  if [[ "$has_initial_branch_option" == 0 ]]; then
+    git init --initial-branch=main >> "$TEST_OUTPUT_FILE" 2>&1
```

Comment:
Why do we set `main` as the default branch?

Reply:
This is to avoid these warnings from git and to avoid the default branch name 'master': 

````
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
git branch -m <name>
````

Reply:
See also https://www.theserverside.com/feature/Why-GitHub-renamed-its-master-branch-to-main

Reply:
Ok 👍